### ISSUE TITLE
Add "make mutable" diagnostic / quick fix for local variables

### ISSUE THREAD

**Issue by jplatte** [rust-lang/rust-analyzer#8449]
Example:

```rust
let foo = Vec::new();

loop {
    // other code
    foo.<|>
}
```

To reproduce:

* Type `pu` => get `Vec::push` suggested
* Accept suggestion => line changes to `foo.push(value)`

Desired outcome: `let foo = Vec::new();` is changed to `let mut foo = Vec::new();` when accepting the suggestion
Actual outcome: `let foo = Vec::new();` is not changed

I would love if ra did this because that would require less jumping around the code.

**Comment by Veykril** [comment]
This sounds neat but I think if we do this it should certainly be behind a config flag since the entire point of explicit mutability is to acknowledge it and I can see some people maybe not wanting this behaviour.

**Comment by matklad** [comment]
I don't think completion is the right place for this. This should be a quick fix for the subsequent diagnostic instead.

**Comment by lnicola** [comment]
#10658 has more info.

**Comment by HKalbasi** [comment]
The diagnostic is implemented in #14232 so I'm going to close this issue. About auto inserting `mut`, I think we should have a general fix-on-format or fix-on-save for diagnostics.
